### ISSUE TITLE
Uncaught ArgumentCountError when trying to print a list of custom DataObject

### ISSUE THREAD

**Issue by jmfederico** [silverstripe/silverstripe-cms#2299]
**silverstripe/cms: 4.2.1**

I have created a custom Versioned DataObject. It has nothing fancy in it. I integrated it with the CMS, and I can create, edit, publish and archive objects without problems. But I can not print the list.

These are the model and admin classes:

```php
<?php

use SilverStripe\Admin\ModelAdmin;

class QuoteAdmin extends ModelAdmin
{
    private static $managed_models = [
        'Aguayo\Quote'
    ];

    private static $url_segment = 'quotes';

    private static $menu_title = 'Quotes';
}
```

```php
<?php

namespace Aguayo;

use SilverStripe\ORM\DataObject;
use SilverStripe\Versioned\Versioned;
use SilverStripe\Assets\Image;

class Quote extends DataObject
{
    private static $db = [
        'quote' => 'HTMLText',
        'person' => 'Varchar',
        'location' => 'Varchar',
        'company' => 'Varchar',
    ];

    private static $has_one = [
        'logo' => Image::class,
    ];

    private static $extensions = [
        Versioned::class
    ];
}
```

When I try to print the list of objects from the CMS, I get the following error:

`[Emergency] Uncaught ArgumentCountError: Too few arguments to function SilverStripe\ORM\FieldType\DBPrimaryKey::__construct(), 0 passed and at least 1 expected`

**Comment by bergice** [comment]
This issue looks to already have been resolved with `4.3.x`, could you try upgrading and see if this resolves it?

**Comment by jmfederico** [comment]
It has been fixed for version `4.3.x`.
